Sharadha taking over for Manvitha-Linkedin-Autoposter

myeeli opened the pull request:
# Description
<img width="575" alt="image" src="https://github.com/user-attachments/assets/fdfd75a8-cda4-435c-897e-4629ca6843dd" />

https://docs.google.com/document/d/1raNHsAl2QMHm1w-3OmzersktXbl4_gyB865xL-y_NYQ/edit?tab=t.0

## Related PRS (if any):
This frontend PR is related to this backend PR:
https://github.com/OneCommunityGlobal/HGNRest/pull/1454
…

## Main changes explained:
- Implemented the LinkedIn auto poster feature where user can post anything on demand, schedule post , edit the scheduled posts and delete the scheduled posts
- The data is not saved in the database
- This is a temporary UI
…

## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. Clear site data/cache
4. log as owner user
5. go to dashboard→ Send Emails


## Screenshots or videos of changes:

https://drive.google.com/file/d/1s-P2BQlhnmWsLp90s8k2N0AaSi14ML66/view?usp=sharing


## Note:
Include the information the reviewers need to know.

myeeli committed:
Manvitha-Linkedin-Autoposter

myeeli committed:
Merge Conflicts

myeeli committed:
Resolve Merge Conflicts

akshith312 reviewed (commented):
At first I encountered this error. I changed the connection strings. The back end ran fine.
But then I encountered an error while starting the front end.
<img width="1472" height="405" alt="image" src="https://github.com/user-attachments/assets/0064fe38-6e31-45da-92aa-01311c2c6496" />
<img width="1143" height="355" alt="image" src="https://github.com/user-attachments/assets/b909be10-c607-4d83-94b2-fc760442a779" />

Please let me know if I am doing any mistake while testing this.

sanjeev29 commented:
Tested the PR changes. The tagged backend [PR - 1454](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/1454) was deleted. I'm testing this with the `development` branch on the backend. Getting a 404 error when I click on `Post Now`.
<img width="1438" height="850" alt="Screenshot 2025-10-17 at 5 43 33 PM" src="https://github.com/user-attachments/assets/1f30ebbb-ea31-4777-a513-7687d8c031fe" />

sumedhkumar96 reviewed (dismissed):
**Tried to test this by running the frontend and backend in my local but the LinkedIn post fails with a 404. I see that the URL localhost:4500 is hardcoded in the code and that's the reason for 404. Please provide complete instructions to test this.**

<img width="1920" height="1030" alt="Screenshot 2025-10-18 181219" src="https://github.com/user-attachments/assets/43ec1eaf-71fb-4738-a229-14b9b287cc93" />

apoorvajainrp21 commented:
I am getting the following error: 
<img width="1870" height="804" alt="Screenshot 2025-10-19 214433" src="https://github.com/user-attachments/assets/9e5b1446-10d9-468d-bd3e-acd8f50a4c33" />

tomkkl commented:
Hi Manvitha,

Testing this does not take me to the /announcements and it makes me stuck at /dashboard. I have attached an image down below. Hope this helps!
<img width="954" height="713" alt="Screenshot 2025-10-25 at 6 18 22 PM" src="https://github.com/user-attachments/assets/f2b34004-d342-485a-b28b-5253313689cd" />

Vinay944924 commented:
Hi Manvitha,

I tried to post a blog in linkedin but it was throwing an error and I have provided the screenshot below. Please let me know if I am doing anything wrong.

<img width="1470" height="825" alt="Screenshot 2025-11-01 at 1 24 20 PM" src="https://github.com/user-attachments/assets/d72a1cd6-5aca-4ea4-aa12-6f03e6f59b7f" />

ManeeshBuddha21 commented:
Tested locally. The LinkedIn Post Editor loaded fine under Dashboard -> Send Emails.
Able to create and schedule posts from the UI, and the layout looks clean.
Tried posting sample text and an image , UI responded correctly but returned “Failed to post to LinkedIn”
<img width="2801" height="1631" alt="21" src="https://github.com/user-attachments/assets/51b02c53-8865-4845-bb62-b690c34c6686" />

debadyuti23 reviewed (dismissed):
Hi @myeeli I have tested the PR with the instructions you provided in this PR and for the auth tokens in backend PR, however it is resulting unauthorized as shown in the [video](https://drive.google.com/file/d/1_xYjJ5X0keOwDWJQiOazyOaeiNE-fSHd/view?usp=sharing).
I have done multiple refreshes and installation resets still the issue persists. I even have tested the linkedin api with same creds, it is working fine. 
Additionally, I have encountered this issue in backend, make sure your feature is synced with recent changes.
<img width="1217" height="714" alt="PR_3647_backend_error_log" src="https://github.com/user-attachments/assets/7e14132f-2fde-405e-9d9d-ae8a1c87d544" />
Therefore please help me understand if I am missing something in testing.
Another thing I noticed that backend api `localhost:4500` is hardcoded, it should be changed

Anusha-Gali reviewed (dismissed):
Hi Manvitha,

I tried reviewing your PR locally and getting the same issue as other reviewers. Kindly look into it.
Even in dark mode we cannot see the text we have typed and the date or time selected.
<img width="1273" height="754" alt="Screenshot 2026-01-06 at 10 49 50 PM" src="https://github.com/user-attachments/assets/b7deed8f-851a-4df3-a91f-978a99ea155f" />
<img width="1273" height="754" alt="Screenshot 2026-01-06 at 10 51 35 PM" src="https://github.com/user-attachments/assets/e18c3db6-38ef-47e9-b015-46d8d2b8c0c2" />

Vikas-8055 reviewed (commented):
Hi Manvitha, I tested this PR along with backend PR #1454. The LinkedIn Post Editor UI loads correctly and text entry works, but clicking "Post Now" fails with the error "Failed to post to LinkedIn".

<img width="1920" height="996" alt="512925691-ff1542aa-bdc7-40be-ae7f-8676d414bb22" src="https://github.com/user-attachments/assets/3df43ab0-2b2f-4fab-bfee-e7e88a92bfd5" />

Shravan-neelamsetty reviewed (commented):
Hi @myeeli, I have tested this PR locally by checking out the Manvitha-Linkedin-AutoPoster branch and navigating to /announcements. The LinkedIn Post Editor UI loads correctly and displays all expected components including the text editor, upload media button, date/time picker, "Post Now" button, and "Scheduled Posts" section. However, I immediately encountered the critical error "Failed to post to LinkedIn" displayed in the notification at the top right of the page.
![3647](https://github.com/user-attachments/assets/c0eec44c-d063-4617-9ff6-ecbcf4ac2b54)

Ganesh112001 reviewed (dismissed):
Tested PR #3647 but cannot access the Announcements page—all navigation attempts redirect to Dashboard even when logged in as Owner. The route appears inaccessible or not properly configured in this branch.

<img width="1470" height="831" alt="Screenshot 2026-01-23 at 2 18 41 AM" src="https://github.com/user-attachments/assets/1d52eed4-e538-4fd9-b5f9-ceb735165abc" />

SharadhaKasiviswanathan commented on src/components/Announcements/index.jsx:
```diff
@@ -92,52 +112,52 @@ function Announcements({ title, email: initialEmail }) {
-    dispatch(
-      sendEmail(emailList.join(','), title ? 'Anniversary congrats' : 'Weekly update', htmlContent),
-    );
+    dispatch(sendEmail(emailList.join(','), 'Weekly Update', htmlContent));
```
Tested the existing email form wiring, and the /sendemail route still passes title and email into Announcements, but this rewritten component no longer accepts those props and always sends with the fixed subject Weekly Update. The prefilled email flow could be checked if the previous EmailPanel behavior needs to stay in place here, as the passed-in email/title are currently ignored.

Copilot committed:
Fix Announcements: add isCanceling state, replace hardcoded URLs with ENDPOINTS, restore email/title props, fix duplicate menubar key

Agent-Logs-Url: https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/sessions/bb774a64-319d-44f1-9c15-27b1ee86403b

Co-authored-by: SharadhaKasiviswanathan <74849358+SharadhaKasiviswanathan@users.noreply.github.com>

Copilot committed:
Fix email input to use controlled value pattern synchronized with emailList state

Agent-Logs-Url: https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/sessions/bb774a64-319d-44f1-9c15-27b1ee86403b

Co-authored-by: SharadhaKasiviswanathan <74849358+SharadhaKasiviswanathan@users.noreply.github.com>

SharadhaKasiviswanathan committed:
Merge remote-tracking branch 'origin/development' into pr-3647-mergecheck

# Conflicts:
#	package-lock.json
#	src/components/Announcements/index.jsx
#	src/utils/URL.js

SharadhaKasiviswanathan committed:
fix(netlify): isolate yarn cache

one-community commented:
Thank you all, merging!
